component/core/src/main/java/org/exoplatform/social/core/metadata/storage/MetadataStorage.java

```diff
@@ -100,6 +100,15 @@ public int deleteMetadataItemsByParentObject(MetadataObject object) {
     return this.metadataItemDAO.deleteMetadataItemsByParentObject(object.getType(), object.getParentId());
   }
 
+  public int deleteMetadataItemsByMetadataTypeAndObject(String metadataTypeName, MetadataObject object) {
+    int deletedItem = 0;
+    List<MetadataItem> metadataItems = this.getMetadataItemsByMetadataTypeAndObject(metadataTypeName, object);
+    for(MetadataItem metadataItem : metadataItems) {
+      deletedItem = this.metadataItemDAO.deleteMetadataItemsByMetadataTypeAndObject(metadataItem.getMetadata().getId(), object.getType(), object.getId());
```

Comment:
```suggestion
      deletedItem =  this.metadataItemDAO.deleteMetadataItemById(metadataItem.getId());
```
Otherwises we should see if it is possible to delete directly metadata items with the given metadata type name and metadata object